windowsforms/matching-game/net45/vb/MatchingGame.Logic/Game.vb

```diff
@@ -0,0 +1,121 @@
+﻿Imports System.Collections.Generic
+
+Public NotInheritable Class Game
+
+    Private ReadOnly _cards() As Char
+    Private ReadOnly _openCards() As Boolean
+    Private ReadOnly _turnCards As New List(Of Integer)(2)
+
+    Private _turns As Integer
+
+    Private Sub New(width As Integer, height As Integer, board As Char())
+        Me.Width = width
+        Me.Height = height
+        _cards = board
+        ReDim _openCards(_cards.Length - 1)
+    End Sub
+
+    Public Shared Function Create() As Game
+        Dim numberOfPairs = 8
+        Dim width = numberOfPairs / 2
+        Dim height = width
+
+        Dim board((width * height) - 1) As Char
+        Dim availableCards = New List(Of Char) From
+        {
+            "!"c, "!"c, "N"c, "N"c, ","c, ","c, "k"c, "k"c,
+            "b"c, "b"c, "v"c, "v"c, "w"c, "w"c, "z"c, "z"c
+        }
+
+        Dim random As New Random()
+
+        For i = 0 To numberOfPairs * 2 - 1
+            Dim cardIndex = random.Next(availableCards.Count - 1)
```

Comment:
The value passed to `Next` is exclusive. Was it meant that `Count - 1` is never generated?
(Applies to both C# and VB code)

Reply:
Unsure. I just used this code from a blog sample done by the team and then converted it. It plays correctly.